harness/tests/experiment/utils.py

```diff
@@ -436,3 +437,29 @@ def make_workloads() -> workload.Stream:
         _ = make_trial_controller_from_trial_implementation(
             trial_class, workloads=make_workloads(), hparams=bad_hparams
         )
+
+
+def assert_patterns_in_logs(input_list: List[str], patterns: List[str]) -> None:
+    """
+    Match each regex pattern in the list to the logs, one-at-a-time, in order.
+    This is adapted from e2e_tests/tests/experiment/experiment.py, which
+    will be moved along with keras e2e_tests.
```

Comment:
Comments in code or docstrings should be time-independent.  Right now this text makes sense because you just moved it, but the average lifespan of a line of code is multiple years, and this docstring won't make any sense 6 months from now.

Git commit messages are a good place for time-aware explanations about decisions which were made, but what goes into the codebase needs to be based on what is physically present right now, not based on previous versions of the code.